hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/OrphanMetaDataManagementTask.java

```diff
@@ -0,0 +1,263 @@
+    final Collection<String> taskTables = getTaskTables();
+    while (eventIterator.hasNext()) {
+      OMDBUpdateEvent<String, ? extends
+          WithObjectID> omdbUpdateEvent = eventIterator.next();
+      OMDBUpdateEvent.OMDBUpdateAction action = omdbUpdateEvent.getAction();
+      // we only process updates on OM's deletedDirectoryTable
+      String table = omdbUpdateEvent.getTable();
+      if (!taskTables.contains(table)) {
+        continue;
+      }
+
+      try {
+        if (table.equals(DELETED_DIR_TABLE)) {
+          // key update on deletedDirectoryTable
+          OMDBUpdateEvent<String, OmKeyInfo> deletedDirTableUpdateEvent =
+              (OMDBUpdateEvent<String, OmKeyInfo>) omdbUpdateEvent;
+          OmKeyInfo updatedKeyInfo = deletedDirTableUpdateEvent.getValue();
+
+          switch (action) {
+          case PUT:
+            handlePutDeleteDirEvent(updatedKeyInfo);
+            break;
+          case DELETE:
+            handleDeleteEvent(updatedKeyInfo.getObjectID(),
+                updatedKeyInfo.getParentObjectID(),
+                updatedKeyInfo.getKeyName());
+            break;
+          case UPDATE:
+            break;
+          default:
+            LOG.debug("Skipping DB update event : {}",
+                omdbUpdateEvent.getAction());
+          }
+        }
+        if (table.equals(BUCKET_TABLE)) {
+          // key update on Bucket Table
+          OMDBUpdateEvent<String, OmBucketInfo> bucketTableUpdateEvent =
+              (OMDBUpdateEvent<String, OmBucketInfo>) omdbUpdateEvent;
+          OmBucketInfo updatedBucketInfo = bucketTableUpdateEvent.getValue();
+
+          switch (action) {
+          case PUT:
+          case UPDATE:
+            break;
+          case DELETE:
+            String dbVolumeKey = reconOMMetadataManager.getVolumeKey(
+                updatedBucketInfo.getVolumeName());
+            OmVolumeArgs volumeArgs =
+                reconOMMetadataManager.getVolumeTable().get(dbVolumeKey);
+            String keyName =
+                StringUtils.join(OM_KEY_PREFIX, volumeArgs.getObjectID(),
+                    OM_KEY_PREFIX, updatedBucketInfo.getObjectID());
+            handleDeleteEvent(updatedBucketInfo.getObjectID(),
+                volumeArgs.getObjectID(), keyName);
+            break;
+          default:
+            LOG.debug("Skipping DB update event : {}",
+                omdbUpdateEvent.getAction());
+          }
+        }
+      } catch (Exception ex) {
+        LOG.error("Unable to process Namespace Summary data in Recon DB. ", ex);
+        return new ImmutablePair<>(getTaskName(), false);
+      }
+    }
+    return new ImmutablePair<>(getTaskName(), true);
+  }
+
+  private void removeOrphanMetaData(long objectID) throws IOException {
+    orphanKeysMetaDataTable.delete(objectID);
+  }
+
+  private void handlePutDeleteDirEvent(OmKeyInfo updatedKeyInfo)
+      throws IOException {
+    long objectID = updatedKeyInfo.getObjectID();
```

Comment:
add comment,
// for cases where parent is not present and moved to deleted table, still sub-files and sub-directory can not be marked as orphans, so its removed from orphan map

Reply:
Ok.